bin/annotate_test_failures

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env ruby
+#
+# Usage:
+#   annotate_test_failures [junit-file-path]
+#
+require 'rexml/document'
+
+junit_path = ARGV.first || File.join('build', 'results', 'report.junit')
+title = ENV.fetch('BUILDKITE_LABEL', 'Tests')
+
+unless File.exist?(junit_path)
+  puts "JUnit file not found: #{junit_path}."
+  exit 0
```

Comment:
> **Note** Exiting with status 0 (no error) is intentional here. We don't want the `annotate_test_failures` command to fail for builds where the "test" step failed for reasons other than a test failure (e.g. a crash of the test runner, or a failure to even launch the app, or a compilation issue…) and which would thus not have created the `.junit` file, because failing on this annotation step in those cases (instead of reporting the earlier failure at a higher level in the caller script) could otherwise be misleading and confusing.

Reply:
Have you considered adding this information to the code?

```suggestion
  # Don't fail if `.junit` file not found to avoid "hiding" the failure in the build that resulted in the that file not being created
  exit 0
```